disco_aws_automation/disco_deploy.py

```diff
@@ -737,3 +677,133 @@ def hostclass_option_default(self, hostclass, key, default=None):
+        '''
+        reason = None
+        hostclass = None
+        previous_ami_id = None
+        status = None
+
+        socify_helper = SocifyHelper(config=self._disco_deploy.config,
+                                     ticket_id=ticket_id,
+                                     dry_run=dry_run,
+                                     command="DeployEvent",
+                                     sub_command=self.sub_command(),
+                                     env=self._disco_deploy.environment_name)
+
+        try:
+            ami = self._get_ami_to_deploy()
+
+            if not ami:
+                reason = "Specified AMI not found:" + str(self._disco_deploy._restrict_amis) if \
+                    self._disco_deploy._restrict_amis \
+                    else "No untested AMIs found."
+                logger.error(reason)
+                status = SocifyHelper.SOC_EVENT_BAD_DATA
+            elif not socify_helper.validate():
+                status = SocifyHelper.SOC_EVENT_BAD_DATA
+                raise RuntimeError("The SOC validation of the associated Ticket and AMI failed.")
+            else:
+                hostclass = DiscoBake.ami_hostclass(ami)
+                previous_ami = self._disco_deploy.get_latest_running_amis().get(hostclass)
+                previous_ami_id = previous_ami.id if previous_ami else ""
+
+                self._deploy_ami(ami, dry_run, deployment_strategy)
+                status = SocifyHelper.SOC_EVENT_OK
+        except Exception as err:
+            socify_helper.send_event(
+                status=status or SocifyHelper.SOC_EVENT_ERROR,
+                ami_id=ami.id if ami else "",
+                hostclass=hostclass or "",
+                message=err.message)
+            raise
+
+        socify_helper.send_event(
+            status=status,
+            ami_id=ami.id if ami else "",
+            hostclass=hostclass or "",
+            previous_ami_id=previous_ami_id or "",
+            message=reason or "")
+
+
+class DiscoDeployTestHelper(DiscoDeployHelperBase):
+    """
+    Disco DeployTestHelper Class implements the logic to associated to the Deploy Test command
+    """
+    def sub_command(self):
+        return 'test'
+
+    def _get_ami_to_deploy(self):
+        """
+        If the ami id is specified using the option --ami then run test on the specified ami
+        independently of its stage,
+        Otherwise use the most recent untested ami for the hostclass
+        """
+        amis = self._disco_deploy.all_stage_amis if self._disco_deploy._restrict_amis else \
+            self._disco_deploy.get_test_amis()
+        return random.choice(amis) if amis else None
+
+    def _deploy_ami(self, ami, dry_run, deployment_strategy):
+        self._disco_deploy.test_ami(ami, dry_run, deployment_strategy)
+
+
+class DiscoDeployUpdateHelper(DiscoDeployHelperBase):
+    """
+    Disco DeployTestHelper Class implements the logic to associated to the Deploy Test command
```

Comment:
Disco DeployTestHelper Class implements the logic to associated to the Deploy **Test** command

This is for the update command